ui/button.reel/button.js

```diff
@@ -83,20 +83,46 @@ var Button = exports.Button = Control.specialize(/** @lends module:"montage/ui/n
             this.needsDraw = true;
         }
     },
-    standardElementTagName: {
-        value: "BUTTON"
-    },
 
-    /* TODO: remove when adding template capability */
     hasTemplate: {
         value: false
     },
+
+    _promise: {
+        value: undefined
+    },
+
+    promise: {
+        get: function () {
+            return this._promise;
+        },
+        set: function (value) {
+            var self = this;
+            var test = function promiseResolved(){
+                            if (promiseResolved.promise === self._promise){
+                                self._promise = undefined;
+                            }
+                        };
+
+            if (this._promise !== value) {
+                this._promise = value;
+
+                if (this._promise){
+                    test.promise = value;
+                    this._promise.then(test); 
+                }
+            }
+        }
+    },
+
     /**
-        converter
         A Montage converter object used to convert or format the label displayed by the Button instance. When a new value is assigned to <code>label</code>, the converter object's <code>convert()</code> method is invoked, passing it the newly assigned label value.
         @type {Property}
         @default null
     */
+    converter: {
```

Comment:
This already exists on Control, please remove the comments as it looks like it got left behind